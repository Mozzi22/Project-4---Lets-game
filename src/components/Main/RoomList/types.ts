export type TRoomListItems = {
    key: string;
    content: string;
    id: string;
}
