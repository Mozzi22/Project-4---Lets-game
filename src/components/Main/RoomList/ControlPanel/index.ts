import { connect } from 'react-redux';
import ControlPanel from './ControlPanel';
// import { rooms } from '/src/Store/chat/selectors';
import { changeModalVisibility } from 'src/Store/modals/actions';

// const mapStateToProps = state => ({ rooms: rooms(state) });
const mapDispatchToProps = dispatch => ({
    changeModalVisibility: payload => dispatch(changeModalVisibility(payload)),
});

export default connect(null, mapDispatchToProps)(ControlPanel);
