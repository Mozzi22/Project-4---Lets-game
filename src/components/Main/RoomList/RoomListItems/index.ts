import RoomListItems from './RoomListItems';

export default RoomListItems;
