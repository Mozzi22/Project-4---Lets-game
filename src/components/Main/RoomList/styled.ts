import styled from 'styled-components';
import { bgColorDefault, bgColorDefaultFon } from 'src/components/UI/baseLayout';

export const StRoomList = styled.div`
    background-color: ${({ color = bgColorDefaultFon }) => color};
    display: flex;
    width: 20%;
    margin-right: 0;
    font-size: 24px;
    flex-direction: column;
    font-family: 'Play', sans-serif;
    border-radius: 0px;
    position: relative;
    height: 90vh;
    border-right: 1px solid black;
`;
export const StControlPanel = styled.div`
    display: flex;
    width: 100%;
    margin-right: 0;
    font-size: 24px;
    flex-direction: column;
    font-family: 'Play', sans-serif;
    border-radius: 0px;
`;
export const StItems = styled.div`
     display: flex;
     height: 68vh;
     flex-direction: column;
     width: 100%;
     overflow-y: hidden;
    :hover {
      overflow-y: auto;
      overflow-x: auto;
    }
        ::-webkit-scrollbar {
          width: 8px;
          height: 8px;
        }
            ::-webkit-scrollbar-thumb {
              width: 8px;
              background-color: ${({ color = bgColorDefault }) => color};
            }
`;
