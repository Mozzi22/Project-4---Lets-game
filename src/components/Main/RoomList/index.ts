import RoomList from './RoomList';
import { connect } from 'react-redux';
import { AplicationState } from 'src/Store/types';
import { getRooms } from 'src/store/game/selectors';

const mapStateToProps = (state: AplicationState) => ({
    rooms: getRooms(state),
});

export default connect(mapStateToProps)(RoomList);
