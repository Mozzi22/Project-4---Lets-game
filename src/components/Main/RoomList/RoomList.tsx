import React from 'react';
import ControlPanel from './ControlPanel';
import RoomListItems from './RoomListItems';
import { StRoomList, StControlPanel, StItems } from './styled';
// import SearchNoRes from './ControlPanel/SearchNoRes';
import { TRoomListItems } from './types';
import { TInitialUserGame, TRoom } from 'src/Store/game/types';

const RoomList  = ({ rooms, userLogin }: TInitialUserGame) => {
    console.log("rooms", rooms, userLogin);
    return (
        <StRoomList>
            <StControlPanel>
                <ControlPanel />
            </StControlPanel>
            <StItems>
                {rooms.map((room: TRoom) => {
                    return (
                        <RoomListItems
                            key={room.id}
                            id={room.id}
                            userLogin={userLogin}
                            gameType={room.gameType}
                            content={room.name}
                        />
                    );
            })}
            </StItems>
        </StRoomList>
    );
};

export default RoomList;
