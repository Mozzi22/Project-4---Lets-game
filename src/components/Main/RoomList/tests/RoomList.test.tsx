import React from 'react';
import RoomList from '../RoomList';
import { mount, shallow } from 'enzyme';

describe('RoomList', () => {
    it('Should match snapshot', () => {
        const component = shallow(<RoomList />);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render StRoomList', () => {
        const component = mount(<RoomList/>);
        expect(component.find('styled__StRoomList')).toHaveLength(1);
    });
     it('should render StControlPanel', () => {
        const component = mount(<RoomList/>);
        expect(component.find('styled__StControlPanel')).toHaveLength(2);
    });
    it('should render StItems', () => {
        const component = mount(<RoomList/>);
        expect(component.find('styled__StItems')).toHaveLength(1);
    });
    it('should render ControlPanel', () => {
        const component = mount(<RoomList/>);
        expect(component.find('ControlPanel')).toHaveLength(1);
    });
    it('should render SearchNoRes', () => {
        const component = mount(<RoomList />);
        expect(component.find('SearchNoRes')).toHaveLength(1);
    });
});
