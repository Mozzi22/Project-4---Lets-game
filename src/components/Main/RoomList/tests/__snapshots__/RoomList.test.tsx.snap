// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomList Should match snapshot 1`] = `"<div class=\\"styled__StRoomList-sc-1u2xsov-0 ekilWs\\"><div class=\\"styled__StControlPanel-sc-1u2xsov-1 jyZVHJ\\"><div class=\\"styled__StControlPanel-sc-17c3c8y-0 cGNAsp\\"><div height=\\"60px\\" class=\\"styled__StInputContainer-sc-153pik-3 bHuQgg\\"><input type=\\"text\\" id=\\"nameOfYourId\\" name=\\"filterByRoomName\\" color=\\"white\\" placeholder=\\"placeholder_search_input\\" class=\\"styled__StInput-sc-153pik-1 ehGRvf\\"/></div><div class=\\"styled__StCreateRoom-sc-17c3c8y-1\\"><button id=\\"create_room\\" name=\\"create_room\\" type=\\"button\\" width=\\"200px\\" class=\\"styled__StButton-sc-15f3n0p-0 fpLSlp\\">create_room</button></div></div></div><div class=\\"styled__StItems-sc-1u2xsov-2 jsuBgQ\\"><div class=\\"styled__StNoResWrapper-sc-dn15ik-0 pTUWO\\"><p class=\\"styled__StNoResTitle-sc-dn15ik-1 JSdxK\\">no_result</p><span class=\\"styled__StNoResText-sc-dn15ik-2 flNXeE\\">try_new</span></div></div></div>"`;
